ttnn/cpp/ttnn/operations/pool/upsample/device/upsample_program_factory_singlecore.cpp

```diff
@@ -39,7 +39,7 @@ operation::ProgramWithCallbacks upsample_single_core(
     // circulat buffer for input
     uint32_t src0_cb_index = CBIndex::c_0;
     uint32_t num_input_units = 2;
-    uint32_t aligned_input_unit_size = round_up_to_mul32(input_unit_size);
+    uint32_t aligned_input_unit_size = round_up_to_mul64(input_unit_size);
```

Comment:
use tt::round_up here for ournd up instead of these functions coming form test file?
Don't just hard code 64 for alignment, read dram alignment from Hal and use that (so on WH you get 32)
I would than close this PR and open  a new one to tigger less folsk (if you remove changes in test_tiles.hpp)